### PR TITLE
Add authorization and key generation module to backend

### PR DESCRIPTION
**Pull request by GToidZ**
This pull request introduces around two new features into the project's backend:
* Authorization using API keys with customizable permission scopes.
* Key generation route on the API.

There is also a gist on how we can use the new functionalities available here: **[Example of how to use check_auth_key and CheckPermissionsOfKey](https://gist.github.com/GToidZ/4c665af77359de3750e3bb8689767d3a)**

### CODE REVIEW

**Review by Halcyon905 — DISMISSED** [review]
seems good to me

File: dump_data/Sota/Keys.metadata.json
Comment (GToidZ) [inline dump_data/Sota/Keys.metadata.json:1]: We still need to fix this file due to issue #2, please block the pull request.

**Review by RyukungG — APPROVED** [review]
Good

**Review by Savetang19 — APPROVED** [review]
LGTM! Approve ka

**Review by MarojT — APPROVED** [review]
Approve

**Review by Halcyon905 — APPROVED** [review]
ok 👍

**Pull request merged**